framework/src/wifi_manager/wifi_manager.c

```diff
@@ -848,6 +862,24 @@ wifi_manager_result_e _handler_on_disconnecting_state(_wifimgr_msg_s *msg)
 }
 
 
+
+wifi_manager_result_e _handler_on_softap_disconnecting_state(_wifimgr_msg_s *msg)
+{
+	WM_LOG_HANDLER_START;
+	if (msg->event == EVT_STA_DISCONNECTED) {
+		if (WIFIMGR_CHECK_CBK) {
+			_handle_user_cb(CB_STA_DISCONNECTED, NULL);
```

Comment:
Maybe user doen't want to receive the disconnect event when it was trying to change mode.

Reply:
An WiFi interface (in our case, it is wl1) can only operate in a single mode. Therefore, when user chagnes WiFi mode from STA to SoftAP, the driver should try disconnection with the associated AP, which causes to trigger disconnection callback passed to WiFi Manager. In case, user wants to operate both SoftAP and STA mode simultaneously (using single WiFi NIC), we may need a virtual WiFi interface additionally.